packages/in_app_purchase/in_app_purchase_storekit/darwin/in_app_purchase_storekit/Sources/in_app_purchase_storekit/StoreKit2/InAppPurchasePlugin+StoreKit2.swift

```diff
@@ -207,12 +207,14 @@ extension InAppPurchasePlugin: InAppPurchase2API {
   /// Sends an transaction back to Dart. Access these transactions with `purchaseStream`
   private func sendTransactionUpdate(transaction: Transaction, receipt: String? = nil) {
     let transactionMessage = transaction.convertToPigeon(receipt: receipt)
-    self.transactionCallbackAPI?.onTransactionsUpdated(newTransactions: [transactionMessage]) {
-      result in
-      switch result {
-      case .success: break
-      case .failure(let error):
-        print("Failed to send transaction updates: \(error)")
+    DispatchQueue.main.async {
```

Comment:
nit: can you do 
```
Task { @MainActor in 
  ....
}
```

To be consistent with other parts of the code?

Reply:
Done. I'm afraid I see tests failing at main (`testGetTransactionJsonRepresentation`, `testFinishTransaction`, at bfb42a62a56ee53010e7c2601cbf1bac993122f2). I assume it'll get sorted out.